include/SVF-FE/LLVMUtil.h

```diff
@@ -104,6 +104,15 @@ inline bool isHeapAllocExtFunViaArg(const SVFFunction* fun)
     return fun && ExtAPI::getExtAPI()->is_arg_alloc(fun);
 }
 
+static inline Type *getPtrElementType(const PointerType* pty) {
+#if (LLVM_VERSION_MAJOR < 14)
+    return getPtrElementType(pty);
+#else
+    assert(!pty->isOpaque() && "Opaque Pointer is used, please recompile the source adding '-Xclang -no-opaque-pointer'");
+    return pty->getNonOpaquePointerElementType();
```

Comment:
Do we have any other way to return a correct type rather than an assertion here?

Reply:
With opaque pointers introduced in llvm-14, pointer type does not contain  element type any more. If you need the element type info, you need to compile the source and disable opaque pointer feature. Please read the following link:

https://llvm.org/docs/OpaquePointers.html

Reply:
I see. Thanks for the explanation.